solidity/contracts/BancorNetwork.sol

```diff
@@ -636,70 +642,58 @@ contract BancorNetwork is IBancorNetwork, TokenHolder, ContractRegistryClient, R
 
     bytes4 private constant GET_RETURN_FUNC_SELECTOR = bytes4(keccak256("getReturn(address,address,uint256)"));
 
-    // using assembly code since older converter versions have different return values
-    function getReturn(address _dest, address _sourceToken, address _targetToken, uint256 _amount) internal view returns (uint256, uint256) {
-        uint256[2] memory ret;
+    // using a static call to get the return from older converters
+    function getReturn(IConverter _dest, IERC20Token _sourceToken, IERC20Token _targetToken, uint256 _amount) internal view returns (uint256, uint256) {
         bytes memory data = abi.encodeWithSelector(GET_RETURN_FUNC_SELECTOR, _sourceToken, _targetToken, _amount);
+        (bool success, bytes memory returnData) = address(_dest).staticcall(data);
+
+        if (success) {
+            if (success && returnData.length == 64) {

```

Comment:
You don't need to check for `success` again

Reply:
oops, guess I missed it in the copy/paste

Reply:
fixed